pkg/api/variable.go

```diff
@@ -18,10 +18,20 @@ func (c *client) ListVariables(ctx context.Context, serviceID string, environmen
 	})
 
 	if err != nil {
-		return nil, err
+		return nil, nil, err
 	}
 
-	return query.Service.Variables, nil
+	variableList := make(model.Variables, 0, len(query.Service.Variables))
+	readonlyVariableList := make(model.Variables, 0, len(query.Service.Variables))
```

Comment:
I do not think this capacity allocation is necessary – it is actually <= len(variables).

Reply:
> I do not think this capacity allocation is necessary – it is actually <= len(variables).

Pre-allocate slice will increase performance, this avoids frequent memory reallocations during the append operation.